crates/sui-adapter-transactional-tests/tests/shared/become_dynamic_object_field.exp

```diff
@@ -33,5 +33,6 @@ Version: 4
 Contents: a::m::Inner {id: sui::object::UID {id: sui::object::ID {bytes: fake(5,0)}}}
 
 task 7 'run'. lines 54-54:
-Error: Transaction Effects Status: Move Runtime Abort. Location: sui::dynamic_field::add_child_object (function index 10) at offset 0, Abort Code: 4
-Debug of error: MoveAbort(MoveLocation { module: ModuleId { address: sui, name: Identifier("dynamic_field") }, function: 10, instruction: 0, function_name: Some("add_child_object") }, 4) at command Some(0)
+created: object(7,0)
```

Comment:
Note that this test now _passes_ instead of fails -- this is now matches the behavior we get with dynamic fields (can add and remove as long as it's within the same transaction).

Reply:
There was a check in `dynamic_object_field` for checking if the object being added was a shared object or not (this was because we couldn't check for wrapping in this case). 

So if you did the add then remove in the transaction this would fail the add. Whereas with a dynamic field you can add then remove in the transaction. This now makes this behavior the same for dynamic object fields.